rfcs/011-dedicated-renku-git-user/011-dedicated-renku-git-user.md

```diff
@@ -0,0 +1,113 @@
+with `Reporter` permissions.
+- KG integration for a specific project can still be verified with the KG. In some cases
+the KG will simply not know anything at all for a project. In this case KG integration
+for this project is not enabled. Checking that the Renkubot user is in the project with the
+right permissions is not enough. It could be that the project has the Renkubot user but
+the KG has not fully finished indexing the project.
+- Enabling KG integration on projects that do not have it will have to be modified to also
+add the user to the project similarly to creating a project.
+- Disabling / removing KG integration results in removal of the project from the KG and removing
+the Renkubot user from the project.
+
+### KG
+
+Depending on when the Gateway refactoring is completed, the KG may not have to deal with the Renkubot
+credentials at all. This is the preferred option. But it may turn out that an initial implementation
+of this gives the Renkubot credentials directly to the KG at first.
+
+## Drawbacks
+
+1. The Renkubot user will have read access to many repos. If the credentials are stolen or leaked
+then they could be used to read the contents of private repos. 
+
+2. When this is rolled out we would automatically add the Renkubot user to all projects. This may
+be confusing to users.
+
+3. Searching for projects and listing projects has to be done either solely through Gitlab or 
+by going to both the KG and Gitlab.
+
+## Rationale and Alternatives
+
+The alternative to this is for the gateway to keep track of which user tokens can be used 
+for which project and then provide these to the KG. This definitely complicates the design
+and implementation of the gateway. If the gateway manages user access tokens for the KG we
+would also have to make the database where the tokens are stored persisten. And losing the
+tokens or letting them expire would mean that the KG would not be able to index projects
+until at least one user for each project with the right permissions logs in.
+
+An interesting benefit to this RFC is that we could get sort of a "naive" KG fedeeration with
+the Renkubot users. For example if a KG for Renku deployment 1 gets the Renkubot credentials 
+for Renku deployment 2, then Renku deployment 1 will be essentially able to generate the same
+graph that is present in Renku deployment 2. Then, users of Renku deployment 1 will be able to 
+search the KG of both deployments. But this is out of scope for this RFC - it is mentioned here
+just as a potential benefit.
+
+## Unresolved Questions
+
+1. If a user is added to a project in Gitlab do they have to click a link in their email
+to accept the invitation? If this is the case then this whole proposal is not feasible.
+
+2. Do we implement simultaneously with the gateway refactoring or do we have an initial
+implementation where the KG handles the Renkubot credentials.
+
+3. Should we add the Renkubot users to existing projects when this is deployed for the 
```

Comment:
We could do that if using the Project Access Tokens we already have allow doing so.